### PR TITLE
Convert Redux actions to TypeScript

### PR DESCRIPTION
**Pull request by eessex**
- Converts all redux actions to ts and adds typings where necessary
- Converts all related tests to ts and updates some to be more explicit/obvious to what they assert
- Adds additional tests for untested functions
- Adds comments for all actions to explain how they are used

### CODE REVIEW

**Comment by damassi** [comment]
At some point we'll want to start getting proper intellisense in for Redux TS code, e.g., 

intellisense

(ignore the different redux project in ^, just for an example)

But that would be best for a different PR and some experimentation. Lots of docs around that here: https://redux.js.org/recipes/usage-with-typescript. For vanilla Redux the setup is a bit verbose :/

**Comment by eessex** [comment]
@damassi yeah definitely interesting for the future, but not in scope here -- am focused on including documentation via comments how actions are used.

File: src/client/actions/test/edit/articleActions.test.ts [hunk 1]
```diff
@@ -36,44 +38,63 @@ describe("articleActions", () => {
...
+        type: "CHANGE_ARTICLE",
+        payload: { data: { title: "Title" } },
+      })
     })
 
     it("Can accept a key/value pair as args", () => {
       changeArticleData("title", "Title")(dispatch, getState)
-      expect(dispatch.mock.calls[0][0].payload.data.title).toBe("Title")
+
+      expect(dispatch).toBeCalledWith({
```
Comment (xtina-starr) [inline src/client/actions/test/edit/articleActions.test.ts]: I'm not sure how this assertion is different from the one above on line 42.

File: src/client/actions/test/edit/articleActions.test.ts [hunk 2]
```diff
@@ -111,52 +134,69 @@ describe("articleActions", () => {
...
-      }, 550)
+      expect(Backbone.sync.mock.calls[0][0]).toBe("update")
     })
   })
 
   describe("#publishArticle", () => {
     let getState
     let dispatch
-    let setArticleSpy = jest.spyOn(Article.prototype, "set")
+    const setArticleSpy = jest.spyOn(Article.prototype, "set")
```
Comment (xtina-starr) [inline src/client/actions/test/edit/articleActions.test.ts]: What's the different in using `spyOn()` versus `fn()` here?

Comment (eessex) [reply src/client/actions/test/edit/articleActions.test.ts]: `spyOn` lets you track calls to the function without overwriting/effecting its implementation

File: src/client/actions/test/edit/articleActions.test.ts (hunk 1 above)
Comment (eessex) [reply src/client/actions/test/edit/articleActions.test.ts]: ahh you are totally correct -- updating now

Comment (eessex) [reply src/client/actions/test/edit/articleActions.test.ts]: This is actually a duplicate test, so going to remove

File: src/client/actions/test/edit/articleActions.test.ts (hunk 2 above)
Comment (xtina-starr) [reply src/client/actions/test/edit/articleActions.test.ts]: Gotcha

**Comment by xtina-starr** [comment]
@eessex feel free to merge on 🍏

**Pull request merged**